### ISSUE TITLE
Strip framework headers during copy-frameworks`

### ISSUE THREAD

**Issue by erichoracek** [Carthage/Carthage#1337]
We have some private/internal frameworks with headers that we would prefer are not copied alongside the framework binary into the app bundle. As far as I can tell, there is not reason why anyone would want to keep headers when embedding frameworks, since they provide no value to the application itself, and take up some extra space in the bundle as well. As such, should `carthage copy-frameworks` strip headers by default? If so, is this something that would be reasonable to attempt in a PR?

**Comment by mdiep** [comment]
> As such, should carthage copy-frameworks strip headers by default?

I don't think so. The space savings are marginal, APIs can be regenerated from the source code, and they can be convenient for determining whether you have the right version of the framework.

I could be convinced to add it as an option to `copy-frameworks`, but I'm not sure whether we want to encourage this generally. :confused:

**Comment by erichoracek** [comment]
As far as I can tell, Xcode does this by default when copying frameworks. To me, it's strange that Carthage would differ from established behavior here. Here's a screenshot that I took on a project that uses Xcode's default "embed frameworks" post-run build phase in place of `carthage copy-frameworks`:

![](https://storage.jumpshare.com/preview/9tEKzN7LxuXwx4fxLnAjDL4okKp-JVafl9XXgvb3GqkqceU-Ms5F17uXQlDhzNwKzvW8WeruKvkLaV2S3noWt38t7wA9n-vlT_jkylN98IjJmZ_gxzDuijCC6OnKLtdF)

Note that there are no headers in the Frameworks directory of the product. This is produced with this configuration:

![](https://storage.jumpshare.com/preview/WtW9A0HA5QMXLdLbBnvsTHcxy3xf5XIGHEuYV0x9DbiDlJVaZfFYpGNttBysqUk_id2yOhcLoofiLMl_dHOXV90Iq-_ZMIwlJNqsu6s4bO0F1kR3dMUjedqC16uBUu85)

With respect to the size of headers being negligible, in our project with 33 dynamic frameworks we have nearly 4MB of headers (before zipping into an `.ipa`). On an iOS app that is downloaded over cellular connections, I'd hardly call that negligible. I found it with the following invocation:
```
$ find /Users/<me>/Library/Developer/Xcode/DerivedData/.../<OurApp.app>/Frameworks -name "*.h" -exec du -ch {} + | grep total$
3.8M	total
```

While I agree that there are reasons that some users would want to include headers in their precompiled binary, it seems like Carthage should attempt to follow the established conventions first, especially when it can affect binary size.
